bevy_lint/src/lints/nursery/duplicate_bevy_dependencies.rs

```diff
@@ -244,14 +244,62 @@ fn minimal_lint(
 /// but [`Version::from_str`] can only parse exact  versions and not ranges.
 fn get_version_from_toml(table: &toml::Value) -> anyhow::Result<Version> {
     match table {
-        toml::Value::String(version) => Version::from_str(version).map_err(anyhow::Error::from),
+        toml::Value::String(version) => Ok(parse_version(version)),
         toml::Value::Table(table) => table
             .get("version")
             .and_then(toml::Value::as_str)
             .ok_or_else(|| anyhow::anyhow!("The 'version' field is required."))
-            .and_then(|version| Version::from_str(version).map_err(anyhow::Error::from)),
+            .map(parse_version),
         _ => Err(anyhow::anyhow!(
             "Unexpected TOML format: expected a toml-string '<crate> = <version>' or a toml-table with '<crate> = {{ version = <version> }} '"
         )),
     }
 }
+
+/// Parse a Version that does not contain any ranges.
+/// In contrast to `cargo_metadata::semver::Version::from_str` this also supports versions in the
+/// format of `1.1` by just setting the patch level to 0.
+fn parse_version(version: &str) -> Version {
+    // split at '-' in order to not include the pre release version in the patch if one is present.
+    let mut iter = version.split('-');
+    // At least a version is always present
+    let mut semver = iter.next().unwrap().split('.');
```

Comment:
Although the `unwrap()` should never fail if the version is well-formed, I don't think we should call it here. Panicking and ICEs don't provide a great UX, so I'd prefer we avoid them for things that are not critical (such as a single lint).

Would it be possible to switch this to emitting a warning / error diagnostic, but still continue other lints? We can save it for a follow-up too, as `duplicate_bevy_dependencies` is still a nursery lint! :)